### PR TITLE
Add support for filtering individual options in help output

### PR DESCRIPTION
**Pull request by Gedochao**
necessary to filter out options tagged as `restricted` or `experimental` in individual sub-commands help in Scala CLI relevant ticket: https://github.com/VirtusLab/scala-cli/issues/1669

### CODE REVIEW

File: core/shared/src/main/scala/caseapp/core/help/HelpFormat.scala [hunk 1]
```diff
@@ -12,6 +13,7 @@ import dataclass._
   sortGroups: Option[Seq[String] => Seq[String]] = None,
   sortedGroups: Option[Seq[String]] = None,
   hiddenGroups: Option[Seq[String]] = None,
+  filterArgs: Option[Arg => Boolean] = None,
```
Comment (alexarchambault) [inline core/shared/src/main/scala/caseapp/core/help/HelpFormat.scala:16]: Better add this at the very end, not to unnecessarily break bin compat.

Comment (alexarchambault) [reply core/shared/src/main/scala/caseapp/core/help/HelpFormat.scala:16]: With a `@since` annotation right before.

Comment (Gedochao) [reply core/shared/src/main/scala/caseapp/core/help/HelpFormat.scala]: should be fixed now

Comment (Gedochao) [reply core/shared/src/main/scala/caseapp/core/help/HelpFormat.scala]: it seems binary compat still breaks on the CI... 🤔

**Review by alexarchambault — APPROVED** [review]

**Pull request merged**